### PR TITLE
Feat: IPFS + Cloudinary integration

### PR DESCRIPTION
**Pull request by kirkas**
Leverage Cloudinary & Pinata public gateway for a wider and more flexible IPFS support.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Review by zencephalon — APPROVED** [review]
LGTM

File: apps/web/app/(basenames)/name/[username]/opengraph-image.tsx [hunk 1]
```diff
@@ -63,24 +68,35 @@ export default async function OpenGraphImage(props: ImageRouteProps) {
...
+      if (ipfsUrl) {
+        imageSource = ipfsUrl;
+      }
+    } else {
       imageSource = avatar;
     }
-  } catch (error) {}
+
+    // Cloudinary resize / fetch
+    imageSource = getCloudinaryMediaUrl({ media: imageSource, format: 'png', width: 120 });
```
Comment (zencephalon) [inline apps/web/app/(basenames)/name/[username]/opengraph-image.tsx:96]: Should we define 120 as a constant somewhere? I see it in more than one place.

File: apps/web/app/(basenames)/name/[username]/opengraph-image.tsx [hunk 2]
```diff
@@ -63,24 +68,35 @@ export default async function OpenGraphImage(props: ImageRouteProps) {
...
   const profilePicture = getBasenameImage(username);
+  const chain = getChainForBasename(username as Basename);
   let imageSource = domainName + profilePicture.src;
 
   // NOTE: Do we want to fail if the name doesn't exists?
   try {
-    const client = getBasenamePublicClient(base.id);
-    const avatar = await client.getEnsAvatar({
+    const client = getBasenamePublicClient(chain.id);
+    let avatar = await client.getEnsText({
```
Comment (zencephalon) [inline apps/web/app/(basenames)/name/[username]/opengraph-image.tsx:77]: Do we ever reassign avatar?

File: apps/web/pages/api/basenames/[name]/assets/cardImage.svg.tsx [hunk 3]
```diff
@@ -37,26 +44,34 @@ export default async function handler(request: NextRequest) {
...
-  const chainId = chainIdFromParams ? Number(chainIdFromParams) : base.id;
+  const chain = getChainForBasename(username as Basename);
   let imageSource = domainName + profilePicture.src;
 
-  // NOTE: Do we want to fail if the name doesn't exists?
   try {
-    const client = getBasenamePublicClient(chainId);
-    const avatar = await client.getEnsAvatar({
+    const client = getBasenamePublicClient(chain.id);
+    let avatar = await client.getEnsText({
```
Comment (zencephalon) [inline apps/web/pages/api/basenames/[name]/assets/cardImage.svg.tsx:52]: Ditto here, how come this is no longer const?

File: apps/web/app/(basenames)/name/[username]/opengraph-image.tsx (hunk 1 above)
Comment (kirkas) [reply apps/web/app/(basenames)/name/[username]/opengraph-image.tsx]: good catch, we actually want a different size for this one :)

**Review by zencephalon — APPROVED** [review]

**Pull request merged**